test/umzug.test.ts

```diff
@@ -78,6 +81,39 @@ describe('alternate migration inputs', () => {
 		});
 	});
 
+	test('up and down functions should not take parameters', async () => {
+		const spy = jest.fn();
+
+		const syncer = fsSyncer(path.join(__dirname, 'generated/umzug/parameterless-fns'), {
+			'migration1.sql': 'select true',
+		});
+		syncer.sync();
+
+		const umzug = new Umzug({
+			migrations: {
+				glob: ['*.sql', { cwd: syncer.baseDir }],
+				resolve: resolveParams => ({
+					...resolveParams,
+					up: async upParams => spy('up', resolveParams.name, upParams.name),
+					down: async downParams => spy('down', resolveParams.name, downParams.name),
+				}),
+			},
+			logger: undefined,
+		});
+
+		await umzug.up();
+
+		expect(spy).toHaveBeenCalledTimes(1);
+		expect(spy).toHaveBeenNthCalledWith(1, 'up', 'migration1.sql', 'migration1.sql');
```

Comment:
Can you also spy and test for `.path` and `.context`, instead of just checking `.name`?